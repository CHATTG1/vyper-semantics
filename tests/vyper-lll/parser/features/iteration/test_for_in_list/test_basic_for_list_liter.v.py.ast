%pgm(, ,
  %fdecl(%@public, data, , %num,
    %forlist(i, %list(3 5 7 9),
      %if(%compareop(%gt, %var(i), 5),
        %return(%var(i))))
    %return(-1))
)
