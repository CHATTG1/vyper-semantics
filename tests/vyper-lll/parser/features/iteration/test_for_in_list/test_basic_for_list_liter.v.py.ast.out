$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 1943314746 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $with ( $_L , 384 , $%seq ( $mstore ( $add ( $mul ( 32 , $uclamplt ( 0 , 4 ) ) , $_L ) , 3 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 1 , 4 ) ) , $_L ) , 5 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 2 , 4 ) ) , $_L ) , 7 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 3 , 4 ) ) , $_L ) , 9 ) ) ) , $repeat ( 352 , 0 , 4 , $%seq ( $mstore ( 320 , $mload ( $add ( 384 , $mul ( $mload ( 352 ) , 32 ) ) ) ) , $%seq ( $if ( $sgt ( $mload ( 320 ) , 5 ) , $%seq ( $%seq ( $mstore ( 0 , $mload ( 320 ) ) , $return ( 0 , 32 ) ) ) ) ) ) ) ) , $%seq ( $mstore ( 0 , -1 ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
